node/overseer/overseer-gen/proc-macro/src/impl_builder.rs

```diff
@@ -19,63 +19,20 @@ use syn::Ident;
 
 use super::*;
 
-/// Returns all combinations for a single replacement:
-/// 1. generic args with `NEW` in place
-/// 2. subsystem type to be replaced
-/// 3. the subsystem name to be replaced by a new type and value
-/// 4. all other subsystems that are supposed to be kept
-fn derive_replacable_generic_lists(
-	info: &OverseerInfo,
-) -> Vec<(TokenStream, Ident, Ident, Vec<Ident>)> {
-	// subsystem generic types
-	let builder_generic_ty = info.builder_generic_types();
-
-	let to_be_replaced_name = info.subsystem_names_without_wip();
-	let baggage_generic_ty = &info.baggage_generic_types();
-
-	builder_generic_ty
-		.iter()
-		.enumerate()
-		.map(|(idx, to_be_replaced_ty)| {
-			let mut to_keep_name = to_be_replaced_name.clone();
-			let to_be_replaced_name: Ident = to_keep_name.remove(idx);
-
-			let mut builder_generic_ty = builder_generic_ty.clone();
-			builder_generic_ty[idx] = format_ident!("NEW");
-
-			let generics_ts = quote! {
-				<S, #( #baggage_generic_ty, )* #( #builder_generic_ty, )* >
-			};
-
-			(generics_ts, to_be_replaced_ty.clone(), to_be_replaced_name, to_keep_name)
-		})
-		.collect::<Vec<(_, _, _, _)>>()
-}
```

Comment:
:+1: this was far from perfect to begin with, but I think it could have been reworked directly to return a `syn::Generics` which is what it generates after all